### ISSUE TITLE
undocumented peer dependencies (babel)

### ISSUE THREAD

**Issue by DanielRuf** [emotion-js/emotion#2921]
**Current behavior:**

When we install projects with pnpm then we get an error due to missing peer dependencies that are not mentioned in your documentation:

![image](https://user-images.githubusercontent.com/827205/195567072-1ef7badf-bac2-495b-acc2-4450bf3ecb1e.png)

https://emotion.sh/docs/introduction#react

![image](https://user-images.githubusercontent.com/827205/195567206-1101bb67-5067-4fb9-b0f3-fc69e032dbc4.png)

**To reproduce:**

Install `@emotion/react` via pnpm instead of yarn or npm.

**Expected behavior:**

**Environment information:**

- `react` version:
- `@emotion/react` version: 11.10.4

**Comment by ArnaudBarre** [comment]
Re https://github.com/emotion-js/emotion/issues/2660

**Comment by Andarist** [comment]
Could you create a repro case for this problem? We declare those as optional peer dependencies and `pnpm` should support those since 2018: https://github.com/pnpm/pnpm/issues/1486

**Comment by DanielRuf** [comment]
> Could you create a repro case for this problem?

Please see my initial comment / post:

> Install `@emotion/react` via pnpm instead of yarn or npm.

Besides that, I do not know if there is a regression or if the other packages like `babel-plugin` are the issue. The link pointing to a similar issue a few months ago is about the exact same issue (as it seems).

**Comment by Andarist** [comment]
Could you open a new issue in pnpm about this then? I see that there are existing tests for optional peer deps in their codebase so I don't know why exactly this happens to `@emotion/react`.

**Comment by ArnaudBarre** [comment]
Yeah that's strange that `pnpm` doens't use this metadata: https://github.com/emotion-js/emotion/blob/main/packages/react/package.json#L80

**Comment by ArnaudBarre** [comment]
There will be an issue anyway with the babel plugin that is a direct dependency and don't mark @babel/core as optional: https://github.com/emotion-js/emotion/blob/main/packages/babel-plugin/package.json#L34

**Comment by Andarist** [comment]
@ArnaudBarre ye, I was just about to say that. We could mark `@babel/core` as optional in `@emotion/babel-plugin` to workaround this issue. It wouldn't be 100% correct for that package... but it would probably be "good enough".

**Comment by ArnaudBarre** [comment]
This would maybe need to be done on babel-plugin-jsx?

With nextjs going full SWC and Emotion working with vite-plugin-react-swc in [the next release](https://github.com/vitejs/vite-plugin-react-swc/commit/6f3e372acc85c2a5663ac6e807cfca68e1c0ca92#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5), it would be great to be able to pull only required dev dependencies and have an `SWC` setup and a `Babel` setup

**Comment by Andarist** [comment]
Could you rephrase? I'm not sure if I fully understand everything that you meant to express.

**Comment by Andarist** [comment]
Could you rephrase? I'm not sure if I fully understand everything that you meant to express.

**Comment by ArnaudBarre** [comment]
I don't at all how Babel is integrated with `@emotion/react` and how complicated this would be but what could be useful is to make `@emotion/react` independent of Babel, and letting the user choose if they want to install Babel plugins or SWC plugins to handle the build-time transformations.

This will allow developers that uses Next 13 or future versions of `@vitejs/plugin-react-swc` to not download any code related to Babel.

**Comment by Andarist** [comment]
> This will allow developers that uses Next 13 or future versions of @vitejs/plugin-react-swc to not download any code related to Babel.

That was an idea behind the current setup with optional peer deps.

> In more concrete terms: Is it possible to make @emotion/babel-plugin an optional peer dependency?

In theory - yes, but in practice this can be seen as a breaking change and I'm hesitant to introduce such a change in v11.

**Comment by michaelgmiller1** [comment]
> In theory - yes, but in practice this can be seen as a breaking change and I'm hesitant to introduce such a change in v11.

Not the most urgent, but would be great to see `@emotion/babel-plugin` be an optional dependency in the next version! It's annoying to see an unmet peer dependency every time you install packages!
